uf2.go

```diff
@@ -0,0 +1,135 @@
+// Converts firmware files from BIN to UF2 format before flashing.
+//
+// For more information about the UF2 firmware file format, please see:
+// https://github.com/Microsoft/uf2
+//
+//
+package main
+
+import (
+	"bytes"
+	"encoding/binary"
+	"io/ioutil"
+)
+
+// ConvertELFFileToUF2File converts an ELF file to a UF2 file.
+func ConvertELFFileToUF2File(infile, outfile string) error {
+	// Read the .text segment.
+	_, data, err := ExtractTextSegment(infile)
+	if err != nil {
+		return err
+	}
+
+	output, _ := ConvertBinToUF2(data)
+	err = ioutil.WriteFile(outfile, output, 0644)
```

Comment:
You could perhaps even simplify this to:

```go
return ioutil.WriteFile(outfile, output, 0644)
```

Reply:
But this is a small nit, I'll merge soon unless you want to fix this.

Reply:
Every line of code removed is a good line of code to remove. Done.